File: model/lyrics.go
package model

import (
	"github.com/allape/gocrud"
)

type Lyrics struct {
	gocrud.Base
	Name        string `json:"name"`
	Index       int32  `json:"index" gorm:"default:0"`
	Content     string `json:"content"`
	Description string `json:"description"`
}

File: ui/src/model/lyrics.ts
import { IBase, IBaseSearchParams } from "@allape/gocrud";

export interface ILyrics extends IBase {
  name: string;
  content: string;
  description: string;
  index: number;
}

export interface ILyricsSearchParams extends IBaseSearchParams {
  like_name?: string;
  in_id?: ILyrics["id"][];
  orderBy_index?: "desc" | "asc";
  orderBy_createdAt?: "desc" | "asc";
  orderBy_updatedAt?: "desc" | "asc";
}
